Scripts/summaryStatistics/biophysics/growth_rate_analysis.py

```diff
@@ -0,0 +1,247 @@
+# Standard modules
+import math
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import pandas as pd
+import pickle
+import sys
+
+# Installed modules
+import CellModeller
+from kneed import KneeLocator
+from sklearn.linear_model import LinearRegression
+
+# Local/custom modules
+from lje.src.lownerJohnEllipse import plot_ellipse, welzl
+   
+def get_radial_position(cell, centerX, centerY):
+    """
+    Convert cell position to radial coordinates relative to colony centroid
+    
+    @param  cell    cellStates object
+    @param  centerX X-coordinate of colony centroid
+    @param  centerY Y-coordinate of colony centroid
+    @return r       Radial position of cell relative to colony centroid
+    """
+    
+    r = math.sqrt((cell.pos[0] - centerX)**2 + (cell.pos[1] - centerY)**2)
+    return r
+    
+def define_bin_ranges(bin_radius, r):
+    """
+    Define the radial ranges for bins to group cells by.
+    
+    @param  bin_radius  Length of the annuli to group cells by
+    @param  r           Radial distance from centroid (nparray)
+    @return bin_ranges  Bin intervals (distances) to average over (nparray)
+    """
+    # Get upper bound for the colony radius
+    max_r = round(np.max(r))
+    
+    # Creat bin ranges
+    bin_ranges = np.arange(0, math.ceil(max_r), bin_radius) #0-10, 10-20, etc.
+    
+    return bin_ranges
+    
+def bin_by_radius(r, measurement, bin_radius, centerX, centerY):
+    """
+    Place cell measurements into bins based on radial position in a colony
+    
+    @param  r           nparray of radial distances from colony centroid
+    @param  measurement measurement of interest (float)
+    @param  bin_radius  Thickness of band to average over (um)
+    @param  centerX     Center of colony (X-coord)
+    @param  centerY     Center of colony (Y-coord)
+    @return df          dataFrame containing: ['Radius'], ['Measurement'], ['bin']
+    """ 
+    # Assign bins to each r and measurement
+    bin_ranges = define_bin_ranges(bin_radius, r)
+    bin_indices = np.digitize(r, bin_ranges) #Allocate each r to a bin number; first bin = 1 #https://www.adamsmith.haus/python/answers/how-to-put-data-into-bins-using-numpy-in-python
```

Comment:
According to the Pep 8 style, each line of code should contain a maximum of 120 characters. This line has 180 characters